Source/WebCore/dom/Document.cpp

```diff
@@ -9593,13 +9579,23 @@ Element* eventTargetElementForDocument(Document* document)
     return element;
 }
 
+// get(Bounding)ClientRect APIs now returns scaled (=zoomed) rect.
+// The zoom argument will be used to divide the rect, returning an unzoomed rect if passed.
+// https://drafts.csswg.org/css-viewport/#zoom-om
```

Comment:
It might be nice to elaborate on this comment a bit more to explain *why* we need to do something like this because I don't think it's obvious from just looking at it.

I think that will also make the implications of the second part of this change more explicit - that we choose want to return `std::nullopt` so that the the code for determining the rect doesn't remove the scales.